# E-Commerce Sales Data Analysis

description here

## Import Modules and Libraries


In [1]:
# Import modules
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [57]:
# Import data
df = pd.read_csv('ecommerce_sales_analysis.csv')
df.head()

product_id product_name        category   price  review_score  \
0           1    Product_1        Clothing  190.40           1.7   
1           2    Product_2  Home & Kitchen  475.60           3.2   
2           3    Product_3            Toys  367.34           4.5   
3           4    Product_4            Toys  301.34           3.9   
4           5    Product_5           Books   82.23           4.2   

   review_count  sales_month_1  sales_month_2  sales_month_3  sales_month_4  \
0           220            479            449             92            784   
1           903             21            989            861            863   
2           163            348            558            567            143   
3           951            725            678             59             15   
4           220            682            451            649            301   

   sales_month_5  sales_month_6  sales_month_7  sales_month_8  sales_month_9  \
0            604            904            446            603            807   
1            524            128            610            436            176   
2            771            409            290            828            340   
3            937            421            670            933             56   
4            620            293            411            258            854   

   sales_month_10  sales_month_11  sales_month_12  
0             252             695             306  
1             294             772             353  
2             667             267             392  
3             157             168             203  
4             548             770             257

## Cleaning Data and Feature Engineering

In [5]:
# Check data types
df.dtypes

product_id          int64
product_name       object
category           object
price             float64
review_score      float64
review_count        int64
sales_month_1       int64
sales_month_2       int64
sales_month_3       int64
sales_month_4       int64
sales_month_5       int64
sales_month_6       int64
sales_month_7       int64
sales_month_8       int64
sales_month_9       int64
sales_month_10      int64
sales_month_11      int64
sales_month_12      int64
dtype: object

In [4]:
# Check for null values
df.isnull().sum()

product_id        0
product_name      0
category          0
price             0
review_score      0
review_count      0
sales_month_1     0
sales_month_2     0
sales_month_3     0
sales_month_4     0
sales_month_5     0
sales_month_6     0
sales_month_7     0
sales_month_8     0
sales_month_9     0
sales_month_10    0
sales_month_11    0
sales_month_12    0
dtype: int64

In [7]:
# Check for duplicates
df.duplicated().sum()

0

In [23]:
# Check columns of the dataset
df.columns

Index(['product_id', 'product_name', 'category', 'price', 'review_score',
       'review_count', 'sales_month_1', 'sales_month_2', 'sales_month_3',
       'sales_month_4', 'sales_month_5', 'sales_month_6', 'sales_month_7',
       'sales_month_8', 'sales_month_9', 'sales_month_10', 'sales_month_11',
       'sales_month_12'],
      dtype='object')

In [58]:
# Created an "annual sales" column by adding sales values from all 12 months
months = ['sales_month_1', 'sales_month_2', 'sales_month_3',
       'sales_month_4', 'sales_month_5', 'sales_month_6', 'sales_month_7',
       'sales_month_8', 'sales_month_9', 'sales_month_10', 'sales_month_11',
       'sales_month_12']

year = 0

for month in months:
    value = df[month]
    year += value


df['annual_sales'] = year
df

product_id  product_name        category   price  review_score  \
0             1     Product_1        Clothing  190.40           1.7   
1             2     Product_2  Home & Kitchen  475.60           3.2   
2             3     Product_3            Toys  367.34           4.5   
3             4     Product_4            Toys  301.34           3.9   
4             5     Product_5           Books   82.23           4.2   
..          ...           ...             ...     ...           ...   
995         996   Product_996  Home & Kitchen   50.33           3.6   
996         997   Product_997  Home & Kitchen  459.07           4.8   
997         998   Product_998          Sports   72.73           1.3   
998         999   Product_999          Sports  475.37           1.2   
999        1000  Product_1000            Toys  225.77           2.1   

     review_count  sales_month_1  sales_month_2  sales_month_3  sales_month_4  \
0             220            479            449             92            784   
1             903             21            989            861            863   
2             163            348            558            567            143   
3             951            725            678             59             15   
4             220            682            451            649            301   
..            ...            ...            ...            ...            ...   
995           494            488            359            137            787   
996           701             18            906            129             78   
997           287            725            109            193            657   
998           720            196            191            315            622   
999           114            890            903            983            769   

     sales_month_5  sales_month_6  sales_month_7  sales_month_8  \
0              604            904            446            603   
1              524            128            610            436   
2              771            409            290            828   
3              937            421            670            933   
4              620            293            411            258   
..             ...            ...            ...            ...   
995            678            970            282            155   
996             19            110            403            683   
997            215            337            664            476   
998            854            122             65            938   
999            134            704            648            400   

     sales_month_9  sales_month_10  sales_month_11  sales_month_12  \
0              807             252             695             306   
1              176             294             772             353   
2              340             667             267             392   
3               56             157             168             203   
4              854             548             770             257   
..             ...             ...             ...             ...   
995             57             575             634             393   
996            104             858             729             474   
997            265             344             888             654   
998            521             268              60             394   
999            495             839             611             110   

     annual_sales  
0            6421  
1            6027  
2            5580  
3            5022  
4            6094  
..            ...  
995          5515  
996          4511  
997          5527  
998          4546  
999          7486  

[1000 rows x 19 columns]

## Sales Trends and Performance Analysis:

### 1. Which product categories have the highest and lowest sales over the past year?

In [9]:
# Checked for every unique product category
df['category'].unique()

array(['Clothing', 'Home & Kitchen', 'Toys', 'Books', 'Electronics',
       'Health', 'Sports'], dtype=object)

In [79]:
# Found the total sales for each product category over the past year 
categories = df['category'].unique()

prod_cat =[]
prod_sales =[]

for category in categories:
    total_sales = df[df['category'] == category]['annual_sales'].sum()
    prod_cat.append(category)
    prod_sales.append(total_sales)
    
Total_sales_prod_cat_df = pd.DataFrame({
    'Category': prod_cat,
    'Total Annual Sales': prod_sales
})

Total_sales_prod_cat_df


Category  Total Annual Sales
0        Clothing              826536
1  Home & Kitchen              742141
2            Toys              917101
3           Books              938229
4     Electronics              845120
5          Health              834414
6          Sports              916371

In [80]:
# Found the product categories with the highest and lowest sales over the past year
print(Total_sales_prod_cat_df.max())
print(Total_sales_prod_cat_df.min())

Category                Toys
Total Annual Sales    938229
dtype: object
Category               Books
Total Annual Sales    742141
dtype: object


**Conclusion:**
- Products that are categorized as toys had the highest number of sales in the past year at $938,229
- Books had the lowest sales over the past year at $742,141

### 2. What are the top-selling products each month?